libs/ledgerjs/packages/hw-app-eth/src/services/ledger/index.ts

```diff
@@ -5,11 +5,162 @@ import { Interface } from "@ethersproject/abi";
 import {
   LedgerEthTransactionResolution,
   LedgerEthTransactionService,
+  LoadConfig,
 } from "../types";
 import { loadInfosForContractMethod } from "./contracts";
 import { byContractAddressAndChainId, findERC20SignaturesInfo } from "./erc20";
 import { getNFTInfo, loadNftPlugin } from "./nfts";
-import { decodeTxInfo } from "../../utils";
+import { decodeTxInfo, tokenSelectors, nftSelectors } from "../../utils";
+
+type potentialResolutions = {
+  token: boolean | undefined;
+  nft: boolean | undefined;
+  externalPlugins: boolean | undefined;
+};
+
+/**
+ * @ignore for external documentation
+ *
+ * Providing additionnal data for some transactions (Token or NFT related) can enable clear signing
+ * of initially impossible to decode data.
+ * This method will add necessary APDUs to the resolution paramter in order to provide this data to the nano app
+ */
+const getAdditionalDataFoContract = async (
+  resolution: LedgerEthTransactionResolution,
```

Comment:
I'm not a big fan of having resolution that is mutate into a function.
I would like a pure function that return the concatenated value of resolution on `return` and then we inject or spread or whatever on `resolution` in the main function.

Nothing really mandatory but it's a personal preference

Reply:
Yeah you're right, me neither, that's how everything was working before and that's why it's done this way here, because I'm always afraid of changing things too much 😆 

I'll try to see if I can make something nicer quickly !

Reply:
Yeah that's why I think it's not mandatory, if it's work keep it I guess 👍

Reply:
Ok just added a new proposition to prevent mutation of parameters, let me know if you think it's better ! 
(To be honest, I have mixed feelings on this, like you I try to never mutate parameters, but here I feel the solution is uglier than the problem, so I don't know 🤷‍♂️)

Reply:
I kinda agree with both of you, best to avoid mutating whenever possible, but I think I preferred the initial version :smile: